pandas/core/arrays/datetimelike.py

```diff
@@ -547,9 +551,18 @@ def _validate_frequency(cls, index, freq, **kwargs):
         if index.size == 0 or inferred == freq.freqstr:
             return None
 
-        on_freq = cls._generate_range(start=index[0], end=None,
-                                      periods=len(index), freq=freq, **kwargs)
-        if not np.array_equal(index.asi8, on_freq.asi8):
+        try:
+            on_freq = cls._generate_range(start=index[0], end=None,
+                                          periods=len(index), freq=freq,
+                                          **kwargs)
+            assert np.array_equal(index.asi8, on_freq.asi8)
+        except (ValueError, AssertionError) as e:
```

Comment:
what is raising an AssertionError? can we change that to a ValueError

Reply:
Done